conmon-rs/server/src/child_reaper.rs

```diff
@@ -125,22 +124,27 @@ impl ReapableChild {
         }
     }
 
-    fn watch(&self, exit_tx: Sender<i32>) -> Receiver<()> {
+    fn watch(&self) -> (Receiver<()>, Receiver<i32>) {
         let exit_paths = self.exit_paths.clone();
         let pid = self.pid;
-        let (tx, rx) = oneshot::channel();
+        let (killed_tx, killed_rx) = oneshot::channel();
+        let (exit_tx, exit_rx) = oneshot::channel();
 
         task::spawn_blocking(move || {
             let wait_status = waitpid(Pid::from_raw(pid as pid_t), None);
             match wait_status {
                 Ok(status) => {
                     if let WaitStatus::Exited(_, exit_code) = status {
-                        Self::write_to_exit_paths(exit_code, &exit_paths)?;
-
+                        // HACK: keep this before write_to_exit_paths.
```

Comment:
@saschagrunert @rphillips @cgwalters any suggestions here? We probably need a different structure...

Reply:
I think the intention to write the exit paths first and then send to the channel was to indicate that we're done with the overall work. Now there is a chance for a race where someone pulled the exit code but we still have no exit paths written.

Hm, can we somehow use `killed_rx` to send the exit code by using a multi-consumer channel? 
https://docs.rs/crossbeam/0.8.1/crossbeam/channel/index.html

Reply:
crossbeam and tokio don't play nice together https://github.com/tokio-rs/tokio/issues/2447

but we could achieve the same thing with tokio's broadcast  I think

Reply:
done!

Reply:
Can we remove the `HACK` comment now?